tape/src/main/java/com/squareup/tape/ObjectQueue.java

```diff
@@ -2,35 +2,58 @@
 package com.squareup.tape;
 
 import java.io.IOException;
+import java.util.List;
 
 /**
  * A queue of objects.
  *
  * @param <T> The type of queue for the elements.
  */
-public interface ObjectQueue<T> {
+public abstract class ObjectQueue<T> {
 
   /** Returns the number of entries in the queue. */
-  int size();
+  abstract int size();
 
   /** Enqueues an entry that can be processed at any time. */
-  void add(T entry) throws IOException;
+  abstract void add(T entry) throws IOException;
 
   /**
    * Returns the head of the queue, or {@code null} if the queue is empty. Does not modify the
    * queue.
    */
-  T peek() throws IOException;
+  abstract T peek() throws IOException;
```

Comment:
This could also have a default implementation:

```java
return peek(1).get();
```

Seem too wasteful to be terribly useful though.